Source/UI/Controllers/IssuersScreen/MLCardFormIssuerTableViewCell.swift

```diff
@@ -41,11 +44,35 @@ final class MLCardFormIssuerTableViewCell: UITableViewCell {
 extension MLCardFormIssuerTableViewCell {
     func setupCell(with issuer: MLCardFormIssuer, radioButtonOn: Bool) {
         radioButtonOn ? setupRadioButton(radioButtonOn: true) : setupRadioButton(radioButtonOn: false)
-        if let imageUrl = issuer.imageUrl {
-            issuerImageView.setRemoteImage(imageUrl: imageUrl)
-        }
+        checkHasImage(issuer: issuer)
         accessibilityLabel = issuer.name
     }
+    
+    private func checkHasImage(issuer: MLCardFormIssuer) {
+        guard let imageURL = issuer.imageUrl else {
+            titleLabel.text = issuer.name
+            issuerImageView.isHidden = true
+            titleLabel.isHidden = false
+            return
+        }
+        
+        DispatchQueue.global(qos: .background).async {
```

Comment:
Question: Why we need to use a `DispatchQueue.global` here, implementing all the feature of retrieving an image from the remote url again, if on the original code we use `issuerImageView.setRemoteImage(imageUrl: imageUrl)`?

This method `setRemoteImage(imageUrl: imageUrl)` doesn't solve this issue already?

Reply:
About "DispatchQueue.global" I'm converting a url into Data, which if it's on the main thread will crash the user depending on the size of what is delivered. That's why we put it in a secondary thread, then we bring it to the main thread to update the layout.

About "setRemoteImage" it is an extension of UIImageView and when receiving the url of the image it automatically assigns it inside the ImageView, I want to check first if by chance the URL is a valid image to appear one or the other. so i couldn't use this extension

Reply:
I think what you want is to add that functionality to `static func request(imageUrl: String, success: ((UIImage)->Void)?) {`
that way you can continue using the caching system that is in place there... just add a way to return and error. or even adding the uiimage as optional to know that there was no valid image and that you need to default to `self.setupPlaceholder(title: issuer.name)`